File: StorageApi/src/main/kotlin/org/gateway/storageApi/batterySystem/dto/BatterySystemDto.kt
package org.gateway.storageApi.batterySystem.dto

import org.gateway.storageApi.consumerGroup.dto.ConsumerGroupDto
import org.gateway.storageApi.dto.Dto
import org.gateway.utils.battery.enums.SystemStatus

class BatterySystemDto : Dto() {
    var voltage: Double = 0.0
    lateinit var status: SystemStatus
    lateinit var manufacturer: String
    lateinit var serialNumber: String
    var group: ConsumerGroupDto? = null
}

File: internal-frontend/src/app/dto/BatterySystemDto.ts
import {SystemStatus} from './SystemStatus';
import {Dto} from './Dto';
import {ConsumerGroupDto} from "./ConsumerGroupDto";

export class BatterySystemDto extends Dto {
  value = '';
  manufacturer = '';
  serialNumber = '';
  group: ConsumerGroupDto | null = null;
  status: SystemStatus = SystemStatus.ONLINE;
}
